src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -14,17 +20,335 @@ public abstract class TextEncoder
         // and so I think the warning is a bit too aggressive.  
         #pragma warning disable 3011
         [CLSCompliant(false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public unsafe abstract bool TryEncodeUnicodeScalar(int unicodeScalar, char* buffer, int bufferLength, out int numberOfCharactersWritten);
 
         // all subclasses have the same implementation of this method.
         // but this cannot be made virtual, because it will cause a virtual call to Encodes, and it destroys perf, i.e. makes common scenario 2x slower 
         [CLSCompliant(false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public unsafe abstract int FindFirstCharacterToEncode(char* text, int textLength);
         #pragma warning restore
 
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public abstract bool Encodes(int unicodeScalar);
 
         // this could be a field, but I am trying to make the abstraction pure.
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public abstract int MaxOutputCharactersPerInputCharacter { get; }
+
+        public string Encode(string value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException("value");
+            }
+
+            unsafe
+            {
+                fixed (char* valuePointer = value)
+                {
+                    int firstCharacterToEncode = FindFirstCharacterToEncode(valuePointer, value.Length);
+
+                    if (firstCharacterToEncode == -1)
+                    {
+                        return value;
+                    }
+
+                    int bufferSize = MaxOutputCharactersPerInputCharacter * value.Length;
+                    char* wholebuffer = stackalloc char[bufferSize];
```

Comment:
I think 256 is way too small for the typical case (encoding response body). @GrabYourPitchforks, any thoughts?